src/helper/proxy/index.ts

```diff
@@ -29,14 +29,37 @@ interface ProxyFetch {
   (input: string | URL | Request, init?: ProxyRequestInit): Promise<Response>
 }
 
+const propagateRequestHeaders = (headers: Headers, init: ProxyRequestInit | undefined) => {
+  let requestHeaders
+
+  if (Array.isArray(init?.headers)) {
+    requestHeaders = init.headers
+  } else if (init?.headers) {
+    requestHeaders = Object.entries(init.headers)
+  }
+
+  if (requestHeaders) {
+    for (const [name, value] of requestHeaders) {
+      if (value === undefined) {
+        headers.delete(name)
+      } else {
+        headers.append(name, value)
+      }
+    }
+  }
+}
+
 const buildRequestInitFromRequest = (
-  request: Request | undefined
+  request: Request | undefined,
+  requestInit: ProxyRequestInit | undefined
 ): RequestInit & { duplex?: 'half' } => {
   if (!request) {
-    return {}
+    // @ts-expect-error `headers` in `requestInit` is not compatible with HeadersInit
+    return requestInit
```

Comment:
This is what lines 94 and 97 were doing previously..

https://github.com/honojs/hono/blob/e7fbc3e2d60a657f5a330daa59b068073f061d5e/src/helper/proxy/index.ts#L92-L99